make_ancestor.sh

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+
+# usage: make_ancestor.sh base ref
+# PWD must be in the relevant git repo
+# merges base in ref if necessary
+
+set -ex
+
+if [ $# != 2 ]; then >&2 echo Bad argument count; fi
+
+base=$1
+ref=$2
+
+basecommit=$(git rev-parse "$base")
+refcommit=$(git rev-parse "$ref")
+
+wtree=$(mktemp -d)
+
+( git checkout --detach HEAD # if ref is already checked out add worktree errors
```

Comment:
It doesn't as long as `ref` is a commit hash. On the other hand, such a `git checkout` could be a source of problems so what about removing it.

Reply:
We really want to check out the branch, not the commit, because we want to add a merge commit to it.

Reply:
What problems would it cause?

Reply:
>We really want to check out the branch, not the commit, because we want to add a merge commit to it.

Ah right.

>What problems would it cause?

None, I hope. Anyways, this could be removed in a next step when making the repository bare.